### PR TITLE
More docs (and RTD)

### PR DESCRIPTION
**Pull request by EntilZha**
With https://github.com/pytorch/text/pull/179 merged I went ahead and did a manual merge of changes I had from the docs I was working on. Mainly this turned out to be:

* Adding in Readthedocs specific stuff to get this working: https://torch-text.readthedocs.io/en/latest/
* Fixing some warnings from sphinx docs
* Adding a couple doc pages that weren't in the original PR.

The tentative plan is after merging that I'll change RTD to point to this repo instead of my fork.

### CODE REVIEW

**Comment by jekbradbury** [comment]
Is there any chance the URL could be `torchtext.readthedocs.io` to align with the pip package name and `torchvision`?

**Comment by EntilZha** [comment]
Done, now its at https://torchtext.readthedocs.io/en/latest/

I'll update the readme link in a sec.

File: torchtext/data/field.py [hunk 1]
```diff
@@ -130,13 +130,11 @@ class Field(RawField):
...
-            eos_token=None, fix_length=None, tensor_type=torch.LongTensor,
-            preprocessing=None, postprocessing=None, lower=False,
-            tokenize=(lambda s: s.split()), include_lengths=False,
-            batch_first=False, pad_token="<pad>", unk_token="<unk>",
-            pad_first=False):
+    def __init__(self, sequential=True, use_vocab=True, init_token=None, eos_token=None, fix_length=None,
+                 tensor_type=torch.LongTensor, preprocessing=None, postprocessing=None, lower=False,
+                 tokenize=(lambda s: s.split()), include_lengths=False, batch_first=False, pad_token="<pad>",
+                 unk_token="<unk>", pad_first=False):
+        super().__init__(preprocessing, postprocessing)
```
Comment (jekbradbury) [inline torchtext/data/field.py]: This line either shouldn't be present or should use the Python 2 `super(self, Field)` syntax.

**Comment by EntilZha** [comment]
I removed that `super` line since its not relevant for docs, fixed some of the lint errors, and merged changes associated with moving from md to rst for the readme (badge). I think the PR should be good to go

**Comment by EntilZha** [comment]
Last thing is that I changed the git project RTD monitors to this one. The last step would be to add the webhook on github which is readthedocs.org/api/v2/webhook/torchtext/22503/. Once that is done it should start auto-building the docs on each commit.

**Comment by jekbradbury** [comment]
Added the webhook 👍; let me know if it's not working

**Pull request merged**